src/index.js

```diff
@@ -219,12 +219,11 @@ class CID {
    */
   static isCID (other) {
     try {
```

Comment:
why is this `try` block still here? shouldn't this be just `return Boolen(CID._checkCIDComponents(other))` now?

Reply:
@dignifiedquire If you look at the original implementation of `validateCID()`, it relies on the behavior of `mh.validate(other.multihash)`, which is supposed to throw an error for invalid multihash. That behavior is preserved in the updated `validateCID()`. And, in order to be able to use/share a common implementation between `isCID()` and `validateCID()`, I had to preserve that behavior in `_checkCIDComponents()` as well. Therefore `isCID()` still needs to catch the error to return false/true _in case_ an error is thrown (e.g., from `mh.validate(other.multihash)`). Contractually (in terms of interface), `isCID()` never throws an error since we now have an "outer" `try/catch` containing its implementation (even through future updates). I hope my answer makes sense.

Reply:
Ah I forgot that. This is unfortunate, `mh.validate` should at least have a counter part that doesn't throw.

Reply:
Yes. but, as I indicated in my comment on #39, the try/catch itself is not an issue. It's the cost of creating an error object in the heap that takes time. Depending on the use cases, _if_ we rarely hit `mh.validate` part in the "if chaining", the cost of having that is not much. The question is, out of 100 times or a million times calling `isCID()` in a typical use case (e.g., are CIDs generally expected to be valid in most cases?, etc.), how many times we end up creating the error object? Having said that, I do agree with you. The code will be cleaner if we don't have to include `try/catch`. Thanks.